### ISSUE TITLE
Add the ability to define custom script aliases

### ISSUE THREAD

**Issue by TravisCarden** [composer/composer#11420]
**tl;dr:** Is there a way to set actual aliases for custom scripts? If not, I propose adding one.

---

### Motivation

I would like to add aliases for my custom Composer scripts, like core commands have, to eliminate redundancy and simplify the command list. For example, `composer prohibits` has an alias of `composer why-not`:

```
$ composer list | grep why-not
  prohibits            [why-not] Shows which packages prevent the given package from being installed
```

As far as I can tell, the only way for me to accomplish something similar for my custom scripts is to basically duplicate them:

```json
{
    "scripts": {
        "phpcs": "phpcs -s",
        "cs": "@phpcs"
    },
    "scripts-descriptions": {
        "phpcs": "Checks code for standards compliance",
        "cs": "Checks code for standards compliance"
    }
}
```

But, besides requiring needless boilerplate, this results in duplication in the command list:

```
$ composer list | grep cs
  cs                   Checks code for standards compliance
  phpcs                Checks code for standards compliance
```

I would, of course, prefer to see this:

```
$ composer list | grep cs
  phpcs                [cs] Checks code for standards compliance
```

### Proposal

Perhaps an approach similar to custom descriptions could be adopted:

```json
{
    "scripts": {
        "phpcs": "phpcs -s"
    },
    "scripts-descriptions": {
        "phpcs": "Checks code for standards compliance"
    },
    "scripts-aliases": {
        "phpcs": ["cs"]
    }
}

```

**Comment by Seldaek** [comment]
The proposal seems reasonable but also not really like something super important (to me), so not something I would personally spend time on.
